### PR TITLE
added button and single input components

### PR DESCRIPTION
**Pull request by jrblaettler**
Added the two requested components. The current input component is only able to do a single input so might just factor it out to be Form and FormInput but we have a single use input now.
Need to add the ability to add CSS to the input but Im gonna do that when I add tailwind and looking at styling in a bigger way.

### CODE REVIEW

**Comment by jochamb** [comment]
i realize i forgot to mention this, but please copy over the tasks i made in the other repository and create a "kanban board" project in this repo -> you can then link pull requests to those tasks and have them auto close as you merge the pull requests

File: components/Button.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+interface ButtonProps {
+  onClick: Function;
+  children: string;
+  className?: string;
+  disabled?: boolean;
+}
+const Button = (props: ButtonProps) => {
+  return (
+    <button className={props.className} onClick={() => props.onClick()} disabled={props.disabled}>
```
Comment (jochamb) [inline components/Button.tsx]:
```typescript
onClick={() => props.onClick()}
```

can be simplified to
```typescript
onClick={props.onClick}
```

File: components/Input.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  };
+  const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    if (props.onSubmit) {
+      props.onSubmit(value);
+    }
+  };
+
+  return (
+    <form onSubmit={handleSubmit}>
```
Comment (jochamb) [inline components/Input.tsx]: i wouldnt have a form element be a part of the `Input` component. even if its only temporary. either remove the `form` from the input or create a resuable `Form` in this pull request and update

**Comment by jrblaettler** [comment]
Will do.

File: components/Input.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      <label htmlFor={props.id}>{props.label}</label>
+      <input
+        id={props.id}
+        name={props.name}
+        type={props.type || 'text'}
```
Comment (jochamb) [inline components/Input.tsx]: the default input type is `text`, so you could omit it if you wanted

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#input_types

File: test/components/Button.test.tsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import Button from '../../components/Button';
```
Comment (jochamb) [inline test/components/Button.test.tsx]: small nit: prefer to stick to module imports instead of using relative

Comment (jochamb) [reply test/components/Button.test.tsx]: nextjs i believe handles this automatically, so can just do `components/Button`

File: test/components/Input.test.tsx [hunk 5]
```diff
@@ -0,0 +1,43 @@
+import Input from '../../components/Input';
```
Comment (jochamb) [inline test/components/Input.test.tsx]: relative import

File: test/components/Input.test.tsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+import { screen, render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import Button from '../../components/Button';
```
Comment (jochamb) [inline test/components/Input.test.tsx]: relative import

**Review by jochamb — CHANGES_REQUESTED** [review]

File: test/components/Input.test.tsx (hunk 5 above)
Comment (jochamb) [reply test/components/Input.test.tsx]: if youre struggling to get it working in jest

https://jestjs.io/docs/configuration#modulenamemapper-objectstring-string--arraystring

```javascript
{
  moduleNameMapper: {
    '^/components(.*)$': '<rootDir>/components$1'
    '^/pages(.*)$': '<rootDir>/pages$1'
  },
}
etc

**Comment by jrblaettler** [comment]
Made the suggested changes

File: components/Button.tsx (hunk 1 above)
Comment (jochamb) [reply components/Button.tsx]: this wasnt changed?

edit: nvm it was, didnt see out of date tag

**Review by jochamb — APPROVED** [review]

**Pull request merged**